api/helpers/get-seasons-without-rankings.js

```diff
@@ -12,14 +12,25 @@ module.exports = {
     }).populateAll();
     return exits.success(
       seasons.map((season) => {
-        return {
+        const res = {
           ...season,
           rankings: new Map(), // playerId => PlayerMatches
           firstPlace: season.firstPlace?.username || null,
           secondPlace: season.secondPlace?.username || null,
           thirdPlace: season.thirdPlace?.username || null,
           fourthPlace: season.fourthPlace?.username || null,
+          gameCounts: [],
+          uniquePlayersPerWeek: [],
         };
+        // initialize gameCounts and uniquePlayersPerWeek
+        const startTime = dayjs(season.startTime);
+        const endTime = dayjs(season.endTime);
+        const numWeeks = Math.ceil(endTime.diff(startTime, 'week', true));
```

Comment:
This computation essentially caused a bug that crashed prod because seasons that were just shorter than a given number of weeks were being truncated here. So Spades 2022 is 11 weeks long except it ended at a time of day 2.5 hours earlier than it started, so without adding `true` to the end of the `.diff()` call, dayjs was truncating and computing it to be 10 weeks long, causing an NPE downstream when we try to compute stats for week 11 (since games did happen that week).

This way we compute the number of weeks in a way that rounds up instead of down, ensuring there's enough weeks represented for all the data associated with that season.

I'd initially implemented a fix in the `StatsController` when we relate a given game to a given week, but I think it's cleaner to do it here. 

Here's my initial writeup of the problem and solution: https://github.com/cuttle-cards/cuttle/pull/481/files#r1285100829